example/app/internal/model/schema_news_test.go

```diff
@@ -756,3 +756,117 @@ func TestNewsRepositoryBase_UpsertQuery(t *testing.T) {
 		})
 	}
 }
+
+func TestNewsRepositoryBaseTx_Insert(t *testing.T) {
+	s := setup(t)
+	defer s.teardown(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 50*time.Second)
+	defer cancel()
+	tx, err := s.news.BeginTx(ctx)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err.Error())
+	}
+
+	var ids []int64
+	for hint, given := range testNewsInsertData {
+		t.Run(hint, func(t *testing.T) {
+			stmtCtx, stmtCancel := context.WithTimeout(ctx, 5*time.Second)
+			defer stmtCancel()
+			ent, err := tx.Insert(stmtCtx, &given.entity)
+			if err != nil {
+				if txErr := tx.Rollback(); txErr != nil {
+					t.Errorf("rollback unexpected error: %s", txErr)
+				}
+				t.Fatalf("unexpected error: %s", err.Error())
+			}
+			ids = append(ids, ent.ID)
+		})
+	}
+	if err := tx.Commit(); err != nil {
+		t.Fatalf("unexpected error: %s", err.Error())
+	}
+
+	for _, id := range ids {
+		if _, err := s.news.FindOneByID(ctx, id); err != nil {
+			t.Fatalf("unexpected error: %s", err.Error())
+		} else {
+			t.Logf("news with id %d found", id)
+		}
+	}
+}
+func TestNewsRepositoryBase_RunInTransaction(t *testing.T) {
```

Comment:
@ders @karthikraobr Here you can find `RunInTransaction` example usage. Let me know what do you think.

Reply:
Looks good to me